### PR TITLE
[server] Set owner token cookie HttpOnly and Secure

### LINKED ISSUE #4830: mark owner token as http only
that it cannot be accessed via document.cookie

### PR DESCRIPTION
**Pull request by csweichel**
Prior the owner token could be read by a VS Code extension or a website served from a port.
Granted, not a very useful thing to do especially in the latter case, because you're already in the workspace then.

fixes #4830

### CODE REVIEW

**Review by akosyakov — APPROVED** [review]
works as advertised

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4832#pullrequestreview-707944876" title="Approved">akosyakov</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/4832#" title="Author self-approved">csweichel</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **jankoehnlein** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @jankoehnlein` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
